File: snippets/java/M08_PassByObjectReference.java
import java.util.*;

class PassByObjectReference {
    static void replace(List<Integer> numbers) {
        numbers = Arrays.asList(42, 43, 44);
    }

    static void append(List<Integer> numbers) {
        numbers.add(42);
    }

    public static void main(String[] args) {
        List<Integer> oneTwoThree = new LinkedList<>(Arrays.asList(1, 2, 3));

        replace(oneTwoThree);
        System.out.println(Arrays.toString(oneTwoThree.toArray()));

        append(oneTwoThree);
        System.out.println(Arrays.toString(oneTwoThree.toArray()));
    }
}

File: snippets/python/m08_pass_by_object_reference.py
def replace(numbers):
    numbers = [42, 43, 44]


def append(numbers):
    numbers.append(42)


one_two_three = [1, 2, 3]

replace(one_two_three)
print(one_two_three)    # > [1, 2, 3]

append(one_two_three)
print(one_two_three)    # > [1, 2, 3, 42]
